datafusion/physical-expr-common/src/intervals/cp_solver.rs

```diff
@@ -0,0 +1,362 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Likewise I think it makes sense to have the interval arithmetic in physical-expr-common too

Reply:
If we do that so, can `Expr`'s still access interval arithmetic?

Reply:
yes

Reply:
If I didn't miss any `pub use` to make it backward compatible, there is no change from others